modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/discovery/NoOperation.scala

```diff
@@ -0,0 +1,89 @@
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.discovery
+
+import scala.concurrent.duration._
+
+import cats.implicits._
+
+import cats.effect.{Timer, Concurrent, Sync}
+
+import fs2.Stream
+import fs2.concurrent.Signal
+
+import eu.timepit.fs2cron.ScheduledStreams
+import eu.timepit.fs2cron.cron4s.Cron4sScheduler
+
+import com.snowplowanalytics.snowplow.rdbloader.config.Config.Schedule
+import com.snowplowanalytics.snowplow.rdbloader.dsl.Logging
+import com.snowplowanalytics.snowplow.rdbloader.state.MakePaused
+import com.snowplowanalytics.snowplow.rdbloader.loading.Load.Status
+
+object NoOperation {
+
+  private implicit val LoggerName: Logging.LoggerName =
+    Logging.LoggerName(getClass.getSimpleName.stripSuffix("$"))
+
+  /**
+   * This is additional timeout duration
+   * Necessary because `waitForIdle` and logging introduce own execution time
+   * and because of that the overall pause time can increase.
+   */
+  val ErrorAllowance: FiniteDuration = 200.millis
+
+  /**
+   * Run a Stream that periodically (specified in schedules.when cron expression)
+   * runs makePaused action, which composed of two actions, first executed immediately
+   * at schedules.when and second executed after schedules.period
+   *
+   * Typically, makePaused first sets global mutable variable `status` to `Paused`,
+   * and after some period sets it back to `Idle`
+   *
+   * The NoOperation stream tries to act only when global status is `Idle` (which means
+   * the Loader is ready for loading). When it's already `Paused` (i.e. two interleaving
+   * schedules) it also does not have an effect until first duration is not expired.
+   */
+  def run[F[_]: Concurrent: Timer: Logging](schedules: List[Schedule], makePaused: MakePaused[F], signal: Signal[F, Status]): Stream[F, Unit] = {
+    val paused = schedules.map {
+      case Schedule(name, cron, duration) =>
+        new ScheduledStreams(Cron4sScheduler.systemDefault[F])
+          .awakeEvery(cron)
+          .evalMap { _ =>
+            // The pause stars only when status signal reaches Idle state
+            val pause = waitForIdle(signal) *>
+              Logging[F].info(s"Transitioning from Idle status for sleeping $duration for $name schedule") *>
+              makePaused(name).use { _ => Timer[F].sleep(duration) } *>
```

Comment:
Hey @spenes! Thanks for looking into this! I think only your second option ("sleep to end at 15.00 pm") is right behavior, i.e. we do this because we know exact timeslot when Loader shouldn't be working (from exactly 14:00 to exactly 15:00).

However, this code should work in exact that way because of `timeoutTo` slightly below. So this `pause` sleeps for an hour since 14:30 (which is wrong), but there's another fiber that sleeps for an hour since 14:00. Whichever finishes sleeping first will unpause the Loader.

We have these two fibers:
1. `pause` makes sure we don't switch non-Idle state
2. `timeoutTo` makes sure we don't pause it after the timeslot

Reply:
Thanks for the explanation @chuwy! I guess I missed that somehow, that totally makes sense.